code/diamm/async_crawler.py

```diff
@@ -0,0 +1,209 @@
+"""
+This script is an asynchronous web crawler that fetches JSON data from the DIAMM website.
+It is highly recommended to use the async version of the crawler instead of the sync version.
+The script will output to a logfile in the working directory, and will also print errors to stderr.
+Change the file handler to log to a different file if needed.
+If you want to revisit pages that have already been visited by previous executions of the crawler,
+set the REVISIT variable to True. This will not prevent it from visiting the same page twice during an execution.
+The script is set to rate limit itself to 40 requests per second, or 1 request every 25ms on average,
+to avoid overwhelming the server.
+"""
+
+import os
+import asyncio
+import aiohttp
+import aiofiles
+import json
+import re
+import logging
+from aiolimiter import AsyncLimiter
```

Comment:
Import order in this and other py files don't currently conform to PEP 8 standard. https://peps.python.org/pep-0008/#imports

A linter will catch this...